packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,148 @@
+
+    /// @notice The minimum number of owners before ownership of the safe is transferred to the fallback owner.
+    uint256 public minOwners;
+
+    /// @notice The fallback owner of the Safe
+    address public fallbackOwner;
+
+    // Constructor to initialize the Safe and baseModule instances
+    constructor(
+        Safe _safe,
+        LivenessGuard _livenessGuard,
+        uint256 _livenessInterval,
+        uint256 _minOwners,
+        address _fallbackOwner
+    ) {
+        safe = _safe;
+        livenessGuard = _livenessGuard;
+        livenessInterval = _livenessInterval;
+        minOwners = _minOwners;
+        fallbackOwner = _fallbackOwner;
+    }
+
+    /// @notice This function can be called by anyone to remove an owner that has not signed a transaction
+    ///         during the livness interval. If the number of owners drops below
+    function removeOwner(address owner) external {
+        // Check that the owner has not signed a transaction in the last 30 days
+        require(
+            livenessGuard.lastSigned(owner) < block.timestamp - livenessInterval,
+            "LivenessModule: owner has signed recently"
+        );
+
+        // Calculate the new threshold
+        address[] memory owners = safe.getOwners();
+        uint256 numOwners = owners.length - 1;
+        uint256 thresholdAfter;
+        if (numOwners > minOwners) {
+            // Preserves the invariant that the Safe has at least 8 owners
+
+            thresholdAfter = get75PercentThreshold(numOwners);
+            console.log("removing one owner. numOwners: %s, thresholdAfter: %s", numOwners, thresholdAfter);
+            safe.execTransactionFromModule({
+                to: address(safe),
+                value: 0,
+                data: abi.encodeCall(
+                    // Call the Safe to remove the owner
+                    OwnerManager.removeOwner,
+                    (getPrevOwner(owner, owners), owner, thresholdAfter)
+                    ),
+                operation: Enum.Operation.Call
+            });
+        } else {
+            console.log("removing all owners. numOwnersAfter: %s", numOwners);
+            // The number of owners is dangerously low, so we wish to transfer the ownership of this Safe to a new
+            // to the fallback owner.
+
+            // The threshold will be 1 because we are removing all owners except the fallback owner
+            thresholdAfter = 1;
+
+            // Remove owners one at a time starting from the last owner.
+            // Since we're removing them in order, the ordering will remain constant,
+            //  and we shouldn't need to query the list of owners again.
+            for (uint256 i = owners.length - 1; i >= 0; i--) {
+                address currentOwner = owners[i];
+                address prevOwner = getPrevOwner(currentOwner, owners);
+                if (currentOwner != address(this)) {
+                    safe.execTransactionFromModule({
```

Comment:
Perhaps wrap `execTransactionFromModule` in an internal function where the values that are the same between all calls are hardcoded, ie the `operation` and the `value` and the `to`

Reply:
+1, I think that'd make this method more readable

Reply:
Good suggestion, we now have `_removeOwner()` and  `_giveToFallbackOwner()` which are much cleaner.